src/main/java/de/tum/in/www1/artemis/service/metrics/MetricsService.java

```diff
@@ -0,0 +1,156 @@
+package de.tum.in.www1.artemis.service.metrics;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+import static de.tum.in.www1.artemis.service.util.ZonedDateTimeUtil.toRelativeTime;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.averagingDouble;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
+import java.time.ZonedDateTime;
+import java.util.Map.Entry;
+import java.util.function.Predicate;
+import java.util.function.ToDoubleFunction;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.repository.metrics.CompetencyMetricsRepository;
+import de.tum.in.www1.artemis.repository.metrics.ExerciseMetricsRepository;
+import de.tum.in.www1.artemis.repository.metrics.LectureUnitMetricsRepository;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyProgressDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.CompetencyStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ExerciseStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitInformationDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.LectureUnitStudentMetricsDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.MapEntryLongLong;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ResourceTimestampDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.ScoreDTO;
+import de.tum.in.www1.artemis.web.rest.dto.metrics.StudentMetricsDTO;
+
+/**
+ * Service class to access metrics regarding students' learning progress.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class MetricsService {
```

Comment:
I'd like you to be more specific in the naming here. Otherwise this may be confused with `MetricsBean` (and related), which is a completely different thing related to monitoring.